src/components/forms/Checkbox.jsx

```diff
@@ -1,78 +1,69 @@
-import React from 'react';
-import uuidv4 from 'uuid/v4';
+import React, { useState } from 'react';
 import PropTypes from 'prop-types';
-import cx from 'classnames';
+import FDS from 'lib/dictionary/js/styleConstants';
 
 import CheckEmptyIcon from 'lib/icons/react/CheckEmptyIcon';
 import CheckFilledIcon from 'lib/icons/react/CheckFilledIcon';
 import CheckIndeterminateIcon from 'lib/icons/react/CheckIndeterminateIcon';
 
-/**
- * @param {Object} props react props
- * @returns {ReactElement}
- */
-const Checkbox = ({
-  label,
-  showLabel,
-  indeterminate,
-  disabled,
-  inputRef,
-  ...otherProps
-}) => {
-  const id = uuidv4();
+import { CustomCheckboxInput, CustomCheckboxContainer } from '@reach/checkbox';
 
-  const IconUnchecked = CheckEmptyIcon;
-  const IconChecked = indeterminate ? CheckIndeterminateIcon : CheckFilledIcon;
+const Checkbox = ({ label, defaultChecked, showLabel = true, ...props }) => {
+  const [checkedState, setChecked] = useState(props.checked || defaultChecked || false);
+  const checked = props.checked != null ? props.checked : checkedState;
+
+  // We should maybe fire onChange in "addition" to setChecked, based on whether we
```

Comment:
not sure if this helps resolve the question, but the main difference to the user is passing `checked` (controlled) vs `defaultChecked` (uncontrolled).